lib/openhab/core/configuration.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module OpenHAB
+  module Core
+    java_import org.openhab.core.config.core.Configuration
+
+    # Adds methods to core openHAB {org.openhab.core.config.core.Configuration} to make it more natural in Ruby
+    class Configuration
+      # inspect result is just the string representation
+      alias_method :inspect, :to_s
+
+      # Return the configuration value for the given key
+      # @!method [](key)
+      # @param [String] key
+      # @return [Object]
+      alias_method :[], :get
+
+      # Check for content equality
+      alias_method :==, :equals
+
+      # Check for the existence of a key
+      # @!method key?(key)
+      # @param [String] key The key to check for existence
+      # @return [Boolean]
+      alias_method :key?, :containsKey
+      alias_method :include?, :key?
+
+      # Convert the {org.openhab.core.config.core.Configuration} object to a {::Hash}
+      # @return [::Hash]
+      def to_h
```

Comment:
should probably also had `to_hash` as an alias (`to_hash` means "this is a hash-like object, that can be implicitly convertible to Hash", whereas `to_h` just means "output a hash, somehow, someway" -- i.e. your usage inside this method takes an array-like object, and has to apply additional logic to get to a hash)